snap/local/hooks/cmd/install/install.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package main
+
+import (
+	"os"
+
+	hooks "github.com/canonical/edgex-snap-hooks/v2"
+	"github.com/canonical/edgex-snap-hooks/v2/env"
+	"github.com/canonical/edgex-snap-hooks/v2/log"
+)
+
+// installProfiles copies the profile configuration.toml files from $SNAP to $SNAP_DATA.
+func installConfig() error {
+	resPath := "/config/device-usb-camera/res"
+	err := os.MkdirAll(env.SnapData+resPath, 0755)
+	if err != nil {
+		return err
+	}
+
+	path := resPath + "/configuration.toml"
+	err = hooks.CopyFile(
+		env.Snap+path,
+		env.SnapData+path)
+	if err != nil {
+		return err
+	}
+
+	path = "/config/rtsp-simple-server.yml"
+	err = hooks.CopyFile(
+		env.Snap+path,
+		env.SnapData+path)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func installDevices() error {
+	devicesDir := "/config/device-usb-camera/res/devices"
+
+	err := os.MkdirAll(env.SnapData+devicesDir, 0755)
+	if err != nil {
+		return err
+	}
+
+	// no default devices
+
+	return nil
+}
+
+func installDevProfiles() error {
+	profilesDir := "/config/device-usb-camera/res/profiles"
+
+	err := os.MkdirAll(env.SnapData+profilesDir, 0755)
+	if err != nil {
+		return err
+	}
+
+	err = hooks.CopyFile(
+		hooks.Snap+profilesDir+"/general.usb.camera.yaml",
```

Comment:
Wouldn't it be better to use wildcards? Otherwise, we'd need to update this hook everytime upstream adds new profiles.

Reply:
All example profiles are copied to the read only snap files and available to the user. The code here is just copying the general one to snap data. 

We may actually be better off not copying any profile/device to snap data because they would be uploaded to core metadata even if not desired. See my other comment: https://github.com/edgexfoundry/device-usb-camera/pull/30#discussion_r879114979

Reply:
Added code to copy the example device to writable directory. I kept the `.example` suffix to avoid unintentional loading. It isn't needed when dynamic discovery is used.